crowbar_framework/app/models/ipmi_service.rb

```diff
@@ -33,6 +33,19 @@ def role_constraints
     end
   end
 
+  def proposal_create_bootstrap(params)
+    if params["deployment"].nil? ||
```

Comment:
DRY. These steps are done in all the services. Can you create a method (i.e. ```set_deployment_element```) ?